### PR TITLE
CORTX-32806: Query Deployment api

### PR DESCRIPTION
**Pull request by lakshita-jain**
Signed-off-by: Lakshita Jain <lakshita.jain@seagate.com>

# Problem Statement
- Query Deployment API code for CSM to utilize for query deployment api which returns the topology as per the requirement of the API using Gconf Data

# Design
Approach followed for generating query deployment topology:

1.  Load input url and copy it to temporary yaml url
2. Remove num_ and unwanted keys from temporary url
3. convert temporary url to dict 
4. process dict data to according to given topology format in HLD.
5. return dict topology.

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: 
-  Confirm All CODACY errors are resolved [Y/N]: 

# Testing
- Are test cases updated along with code changes due to Enhancements/Bugs [Y/N]:
- Confirm that new test cases are added to regression and sanity plan files and relevant feature plan files [Y/N]:
- Confirm that Test Cases are added for new features added [Y/N]: 
- Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: 
- Confirm Testing was performed with installed RPM/K8s deployment [Y/N]:  

# Review Checklist 
####  Before posting the PR please ensure:
- [x] PR is self reviewed
- [x] Is there a change in filename/package/module or signature [Y/N]: 
- [x] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [ ] New package/s added to setup.py?
- [x] Jira is updated
- [x] Check if the description is clear and explained. 
- [x] Check Acceptance Criterion is defined. 
- [x] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified.

# KVstore/Confstore feature (changes/fixes) checklist
#### Confirm changes are made for:
- [ ] ConfStore
- [ ] KVStore
- [ ] ConfCli
- [ ] Test cases added for all above 3
- [ ] changes done in all the KVpayloads (ConsulKvPayload, IniKvPayload, KvPayload)

# Documentation
- [x] Changes done to WIKI / Confluence page

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: py-utils/src/utils/query_deployment/query_deployment.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            nodes_info['machine_id'] = nodes_key
+            for key, val in data['node'][nodes_key].items():
+                if key=='provisioning':
+                    # TODO: uncomment below once deployment time is supported by provisioner
+                    # nodes_info['deployment_time']=data['node'][nodes_key]['provisioning']['time']
+                    nodes_info['version']=data['node'][nodes_key]['provisioning']['version']
+                else:
+                    nodes_info[key]=val
+                #TBD json dumps to dict
+            _config["nodes"].append(json.dumps(nodes_info))
```
Comment (tanujashinde0405) [inline py-utils/src/utils/query_deployment/query_deployment.py:131]: follow wither single quotes or double quotes for variables

File: py-utils/src/utils/query_deployment/query_deployment.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        _f = QueryConfData._local_file
+        if os.path.exists(_f): os.remove(_f)
+
+    def get_data(self, kv_url: str):
+        """Get data related to the parent key from config."""
+        return self._get_data(kv_url)
+
+    def _get_data(self, kv_url: str):
+        """Return data in dict format."""
+        not_required_keys=['num_','nodes']
```
Comment (tanujashinde0405) [inline py-utils/src/utils/query_deployment/query_deployment.py:51]: resolve codacy issues.

File: py-utils/src/utils/query_deployment/query_deployment.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                }
+             },
+             "cluster": [],
+              "nodes": []
+            }
+class QueryConfData:
+    """Query Data."""
+    _query_idx = "query_idx"
+    _data_idx = "data_idx"
+    _local_file = '/tmp/local_conf.conf'
```
Comment (tanujashinde0405) [inline py-utils/src/utils/query_deployment/query_deployment.py:38]: either follow single quotes or double quotes for strings

**Review by tanujashinde0405 — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/query_deployment/query_deployment.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        #cluster info
+        for cluster_key, cluster_val in data['cluster'].items():
+            cluster_info=nd()
+            storage_set_info=nd()
+            storage_set_list=[]
+            cluster_info['security']=data['cortx']['common']['security']
+            if cluster_key=='storage_set':
+                for storage_info in data['cluster']['storage_set']:
+                    for storage_key,storage_val in storage_info.items():
+                        if storage_key!='nodes':
```
Comment (tanujashinde0405) [inline py-utils/src/utils/query_deployment/query_deployment.py:111]: add spaces before and after =

**Review by tanujashinde0405 — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/query_deployment/query_deployment.py [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+        Conf.load(QueryConfData._query_idx, kv_url)
+        _data_keys = Conf.get_keys(QueryConfData._query_idx)
+
+        Conf.load(QueryConfData._data_idx, QueryConfData._local_conf)
+        Conf.copy(QueryConfData._query_idx, QueryConfData._data_idx, _data_keys)
+        Conf.save(QueryConfData._data_idx)
+        for key in Conf.get_keys(QueryConfData._data_idx):
+            if 'num_'in key:
+                Conf.delete(QueryConfData._data_idx, key)
+            # if 'nodes'in key:
```
Comment (tanujashinde0405) [inline py-utils/src/utils/query_deployment/query_deployment.py:61]: remove this commented code if not required

File: py-utils/src/utils/query_deployment/query_deployment.py [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+            raise QueryDeploymentError(errno.EINVAL, f"Invalid data in {kv_url}")
+
+        return QueryDeployment._get_cortx_topology(_data)
+
+    def _get_cortx_topology(data: dict) -> dict:
+        """ Map gconf fields to topology """
+        nd=lambda: defaultdict(nd)
+        _config=Topology.topology
+        
+        #cortx_info
```
Comment (tanujashinde0405) [inline py-utils/src/utils/query_deployment/query_deployment.py:99]: write proper comments on what the code section is fetching

File: py-utils/src/utils/query_deployment/query_deployment.py [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+            else:
+                cluster_info[cluster_key]=cluster_val
+        _config['cluster'].append((json.dumps(cluster_info)))
+
+        # Nodes Info
+        for nodes_key, nodes_value in data['node'].items():
+            nodes_info=nd()
+            nodes_info['machine_id'] = nodes_key
+            for key, val in data['node'][nodes_key].items():
+                if key=='provisioning':
```
Comment (tanujashinde0405) [inline py-utils/src/utils/query_deployment/query_deployment.py:124]: check if such hardcoded values can be added to const.py

**Review by tanujashinde0405 — CHANGES_REQUESTED** [review]

**Review by tanujashinde0405 — APPROVED** [review]
LGTM

File: py-utils/src/utils/query_deployment/query_deployment.py [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+           "cortx": {
+            "common": {
+               "release": {}
+                }
+             },
+            "cluster": [],
+            "nodes": [],
+            }
+class QueryConfData:
+    """Query Data."""
```
Comment (nitin-seagate) [inline py-utils/src/utils/query_deployment/query_deployment.py:35]: Query Configuration Data

File: py-utils/src/utils/query_deployment/query_deployment.py [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+    _query_idx = "query_idx"
+    _data_idx = "data_idx"
+    _local_file = "/tmp/local_conf.conf"
+    _local_conf = "yaml://" + _local_file
+
+    def __init__(self):
+        _f = QueryConfData._local_file
```
Comment (nitin-seagate) [inline py-utils/src/utils/query_deployment/query_deployment.py:42]: We should not delete any file here in init.... Please move this to section where we create file

File: py-utils/src/utils/query_deployment/query_deployment.py [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+
+        Conf.load(QueryConfData._data_idx, QueryConfData._local_conf)
+        Conf.copy(QueryConfData._query_idx, QueryConfData._data_idx, _data_keys)
+        Conf.save(QueryConfData._data_idx)
+        for key in Conf.get_keys(QueryConfData._data_idx):
+            if 'num_'in key:
+                Conf.delete(QueryConfData._data_idx, key)
+        Conf.save(QueryConfData._data_idx)
+
+        from cortx.utils.conf_store import ConfStore
```
Comment (nitin-seagate) [inline py-utils/src/utils/query_deployment/query_deployment.py:62]: Remove this, reuse included header from top...

File: py-utils/src/utils/query_deployment/query_deployment.py [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+        _cs = ConfStore()
+        _cs.load(QueryConfData._data_idx,  QueryConfData._local_conf)
+        _data = _cs.get_data(QueryConfData._data_idx)
```
Comment (nitin-seagate) [inline py-utils/src/utils/query_deployment/query_deployment.py:65]: Just before returning data, unload and cleanup local file

File: py-utils/src/utils/query_deployment/query_deployment.py [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+        _data = QueryDeployment._query_conf.get_data(kv_url)
+        if not len(_data) > 0:
+            raise QueryDeploymentError(errno.EINVAL, f"Invalid data in {kv_url}")
+        return QueryDeployment._get_cortx_topology(_data)
+
+    def _get_cortx_topology(data: dict) -> dict:
+        """ Map gconf fields to topology """
+        nd = lambda: defaultdict(nd)
+        _config = Topology.topology
+        # to fetch cortx info
```
Comment (nitin-seagate) [inline py-utils/src/utils/query_deployment/query_deployment.py:94]: # to fetch common info

File: py-utils/src/utils/query_deployment/query_deployment.py [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+        _config["cortx"]["common"]["release"] = data["cortx"]["common"]["release"]
+        # to fetch cluster_info
```
Comment (nitin-seagate) [inline py-utils/src/utils/query_deployment/query_deployment.py:96]: # to fetch cluster info

File: py-utils/src/utils/query_deployment/query_deployment.py [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+                for storage_info in data['cluster']['storage_set']:
+                    for storage_key,storage_val in storage_info.items():
+                        if storage_key != 'nodes':
+                            storage_set_info[storage_key] = storage_val
+                storage_set_list.append(storage_set_info)
+                cluster_info['storage_set'] = storage_set_list
+            else:
+                cluster_info[cluster_key] = cluster_val
+        _config['cluster'].append((json.dumps(cluster_info)))
+        # to fetch Nodes Info
```
Comment (nitin-seagate) [inline py-utils/src/utils/query_deployment/query_deployment.py:112]: # to fetch nodes info

**Review by nitin-seagate — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/query_deployment/query_deployment.py [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+        if QueryDeployment._query_conf is None:
+            QueryDeployment.init()
+
+        _data = {}
+        _data = QueryDeployment._query_conf.get_data(kv_url)
+        if not len(_data) > 0:
+            raise QueryDeploymentError(errno.EINVAL, f"Invalid data in {kv_url}")
+        return QueryDeployment._get_cortx_topology(_data)
+
+    def _get_cortx_topology(data: dict) -> dict:
```
Comment (tanujashinde0405) [inline py-utils/src/utils/query_deployment/query_deployment.py:90]: please make sure , After deployment API is working and returning correct topology.
Paste the results in PR description.

File: py-utils/src/utils/query_deployment/query_deployment.py [hunk 16]
```diff
@@ -0,0 +1,138 @@
...
+        """ Map gconf fields to topology """
+        nd = lambda: defaultdict(nd)
```
Comment (tanujashinde0405) [inline py-utils/src/utils/query_deployment/query_deployment.py:92]: rename nd

**Review by tanujashinde0405 — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/query_deployment/query_deployment.py [hunk 17]
```diff
@@ -0,0 +1,138 @@
...
+                    # TODO: uncomment below once deployment time is supported by provisioner
+                    # nodes_info['deployment_time'] = data['node'][nodes_key]['provisioning']['time']
+                    nodes_info['version'] = data['node'][nodes_key]['provisioning']['version']
+                else:
+                    nodes_info[key] = val
+                #TBD json dumps to dict
+            _config["nodes"].append(json.dumps(nodes_info))
+        return _config
+
+class QueryDeploymentError(Exception):
```
Comment (tanujashinde0405) [inline py-utils/src/utils/query_deployment/query_deployment.py:127]: move this class to error.py and use utilsError

**Review by tanujashinde0405 — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/query_deployment/query_deployment.py (hunk 9 above)
Comment (tanujashinde0405) [inline py-utils/src/utils/query_deployment/query_deployment.py:42]: move this variable and move the delete code at the end of return data

**Review by tanujashinde0405 — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/query_deployment/query_deployment.py [hunk 18]
```diff
@@ -0,0 +1,138 @@
...
+        Conf.load(QueryConfData._data_idx, QueryConfData._local_conf)
+        Conf.copy(QueryConfData._query_idx, QueryConfData._data_idx, _data_keys)
+        Conf.save(QueryConfData._data_idx)
+        for key in Conf.get_keys(QueryConfData._data_idx):
+            if 'num_'in key:
+                Conf.delete(QueryConfData._data_idx, key)
+        Conf.save(QueryConfData._data_idx)
+
+        from cortx.utils.conf_store import ConfStore
+        _cs = ConfStore()
```
Comment (tanujashinde0405) [inline py-utils/src/utils/query_deployment/query_deployment.py:63]: Lets remove this _cs and directly load _data_idx using Conf which is already imported

File: py-utils/src/utils/query_deployment/query_deployment.py [hunk 19]
```diff
@@ -0,0 +1,138 @@
...
+        if not len(_data) > 0:
+            raise QueryDeploymentError(errno.EINVAL, f"Invalid data in {kv_url}")
+        return QueryDeployment._get_cortx_topology(_data)
+
+    def _get_cortx_topology(data: dict) -> dict:
+        """ Map gconf fields to topology """
+        nd = lambda: defaultdict(nd)
+        _config = Topology.topology
+        # to fetch cortx info
+        _config["cortx"]["common"]["release"] = data["cortx"]["common"]["release"]
```
Comment (tanujashinde0405) [inline py-utils/src/utils/query_deployment/query_deployment.py:95]: replace double quoted with single quotes in data[]

**Review by tanujashinde0405 — CHANGES_REQUESTED** [review]

**Review by tanujashinde0405 — APPROVED** [review]
please make sure , After deployment API is working and returning correct topology.
Paste the results in PR description.

**Comment by lakshita-jain** [comment]
@tanujashinde0405 : here's the output
`>>> from cortx.utils.query_deployment import QueryDeployment
>>> QueryDeployment.get_cortx_topology('consul://cortx-consul-server:8500/conf')
{'cortx': {'common': {'release': {'name': 'CORTX', 'version': '2.0.0-7423'}}}, 'cluster': ['{"security": {"device_certificate": "/etc/cortx/solution/ssl/stx.pem", "domain_certificate": "/etc/cortx/solution/ssl/stx.pem", "ssl_certificate": "/etc/cortx/solution/ssl/s3.seagate.com.pem"}, "storage_set": [{"durability": {"dix": {"data": "1", "parity": "0", "spare": "0"}, "sns": {"data": "1", "parity": "0", "spare": "0"}}, "name": "storage-set-1"}]}'], 'nodes': ['{"machine_id": "3b404b96fc01319b5edb2e84e323595a", "cluster_id": "8ba28deb-4e10-4f07-a32d-ab0d2aed7ffe", "components": [{"name": "utils", "version": "2.0.0-7423"}, {"name": "hare", "version": "2.0.0-7423"}, {"name": "rgw", "services": ["rgw_s3"], "version": "2.0.0-7423"}], "hostname": "cortx-server-1.cortx-server-headless.cortx.svc.cluster.local", "name": "cortx-server-1", "node_id": "cortx-server-1.cortx-server-headless.cortx.svc.cluster.local", "version": "2.0.0-7423", "storage_set": "storage-set-1", "type": "server_node"}', '{"machine_id": "6ac2b08cfe659acf910fed4d11b7df53", "cluster_id": "8ba28deb-4e10-4f07-a32d-ab0d2aed7ffe", "components": [{"name": "utils", "version": "2.0.0-7423"}, {"name": "motr", "services": ["io"], "version": "2.0.0-7423"}, {"name": "hare", "version": "2.0.0-7423"}], "cvg": [{"devices": {"data": ["/dev/sdd", "/dev/sde"], "metadata": ["/dev/sdc"]}, "name": "cvg-01", "type": "ios"}], "hostname": "cortx-data-g0-2.cortx-data-headless.cortx.svc.cluster.local", "name": "cortx-data-g0-2", "node_group": "ssc-vm-g2-rhev4-3385.colo.seagate.com", "node_id": "cortx-data-g0-2.cortx-data-headless.cortx.svc.cluster.local", "version": "2.0.0-7423", "storage_set": "storage-set-1", "type": "data_node/0"}', '{"machine_id": "7ba8990a85d38508ca9bac4ff516d3ea", "cluster_id": "8ba28deb-4e10-4f07-a32d-ab0d2aed7ffe", "components": [{"name": "utils", "version": "2.0.0-7423"}, {"name": "motr", "services": ["io"], "version": "2.0.0-7423"}, {"name": "hare", "version": "2.0.0-7423"}], "cvg": [{"devices": {"data": ["/dev/sdd", "/dev/sde"], "metadata": ["/dev/sdc"]}, "name": "cvg-01", "type": "ios"}], "hostname": "cortx-data-g0-0.cortx-data-headless.cortx.svc.cluster.local", "name": "cortx-data-g0-0", "node_group": "ssc-vm-g2-rhev4-3387.colo.seagate.com", "node_id": "cortx-data-g0-0.cortx-data-headless.cortx.svc.cluster.local", "version": "2.0.0-7423", "storage_set": "storage-set-1", "type": "data_node/0"}', '{"machine_id": "7d9ee991ae6ce628a0e10294bdebaa6b", "cluster_id": "8ba28deb-4e10-4f07-a32d-ab0d2aed7ffe", "components": [{"name": "utils", "version": "2.0.0-7423"}, {"name": "ha", "version": "2.0.0-7423"}], "hostname": "cortx-ha-headless", "name": "cortx-ha", "node_id": "cortx-ha-headless", "version": "2.0.0-7423", "storage_set": "storage-set-1", "type": "ha_node"}', '{"machine_id": "aa378e768a2ba5291d786b3dc48d5650", "cluster_id": "8ba28deb-4e10-4f07-a32d-ab0d2aed7ffe", "components": [{"name": "utils", "version": "2.0.0-7423"}, {"name": "motr", "services": ["io"], "version": "2.0.0-7423"}, {"name": "hare", "version": "2.0.0-7423"}], "cvg": [{"devices": {"data": ["/dev/sdg", "/dev/sdh"], "metadata": ["/dev/sdf"]}, "name": "cvg-02", "type": "ios"}], "hostname": "cortx-data-g1-0.cortx-data-headless.cortx.svc.cluster.local", "name": "cortx-data-g1-0", "node_group": "ssc-vm-g2-rhev4-3387.colo.seagate.com", "node_id": "cortx-data-g1-0.cortx-data-headless.cortx.svc.cluster.local", "version": "2.0.0-7423", "storage_set": "storage-set-1", "type": "data_node/1"}', '{"machine_id": "ac2f9656e38a92990522025e1f3ccaef", "cluster_id": "8ba28deb-4e10-4f07-a32d-ab0d2aed7ffe", "components": [{"name": "utils", "version": "2.0.0-7423"}, {"name": "hare", "version": "2.0.0-7423"}, {"name": "rgw", "services": ["rgw_s3"], "version": "2.0.0-7423"}], "hostname": "cortx-server-0.cortx-server-headless.cortx.svc.cluster.local", "name": "cortx-server-0", "node_id": "cortx-server-0.cortx-server-headless.cortx.svc.cluster.local", "version": "2.0.0-7423", "storage_set": "storage-set-1", "type": "server_node"}', '{"machine_id": "cd1e4c61d297a784245dd88106262cd3", "cluster_id": "8ba28deb-4e10-4f07-a32d-ab0d2aed7ffe", "components": [{"name": "utils", "version": "2.0.0-7423"}, {"name": "motr", "services": ["io"], "version": "2.0.0-7423"}, {"name": "hare", "version": "2.0.0-7423"}], "cvg": [{"devices": {"data": ["/dev/sdd", "/dev/sde"], "metadata": ["/dev/sdc"]}, "name": "cvg-01", "type": "ios"}], "hostname": "cortx-data-g0-1.cortx-data-headless.cortx.svc.cluster.local", "name": "cortx-data-g0-1", "node_group": "ssc-vm-g2-rhev4-3386.colo.seagate.com", "node_id": "cortx-data-g0-1.cortx-data-headless.cortx.svc.cluster.local", "version": "2.0.0-7423", "storage_set": "storage-set-1", "type": "data_node/0"}', '{"machine_id": "d62dc0d2c3ee12aee9b946379519133a", "cluster_id": "8ba28deb-4e10-4f07-a32d-ab0d2aed7ffe", "components": [{"name": "utils", "version": "2.0.0-7423"}, {"name": "hare", "version": "2.0.0-7423"}, {"name": "rgw", "services": ["rgw_s3"], "version": "2.0.0-7423"}], "hostname": "cortx-server-2.cortx-server-headless.cortx.svc.cluster.local", "name": "cortx-server-2", "node_id": "cortx-server-2.cortx-server-headless.cortx.svc.cluster.local", "version": "2.0.0-7423", "storage_set": "storage-set-1", "type": "server_node"}', '{"machine_id": "ecabccb4bced858e282f568d6bf04558", "cluster_id": "8ba28deb-4e10-4f07-a32d-ab0d2aed7ffe", "components": [{"name": "utils", "version": "2.0.0-7423"}, {"name": "csm", "services": ["agent"], "version": "2.0.0-7423"}], "hostname": "cortx-control", "name": "cortx-control", "node_id": "cortx-control", "version": "2.0.0-7423", "storage_set": "storage-set-1", "type": "control_node"}', '{"machine_id": "f488442b875fe8c4328e148f8aca7aa7", "cluster_id": "8ba28deb-4e10-4f07-a32d-ab0d2aed7ffe", "components": [{"name": "utils", "version": "2.0.0-7423"}, {"name": "motr", "services": ["io"], "version": "2.0.0-7423"}, {"name": "hare", "version": "2.0.0-7423"}], "cvg": [{"devices": {"data": ["/dev/sdg", "/dev/sdh"], "metadata": ["/dev/sdf"]}, "name": "cvg-02", "type": "ios"}], "hostname": "cortx-data-g1-2.cortx-data-headless.cortx.svc.cluster.local", "name": "cortx-data-g1-2", "node_group": "ssc-vm-g2-rhev4-3385.colo.seagate.com", "node_id": "cortx-data-g1-2.cortx-data-headless.cortx.svc.cluster.local", "version": "2.0.0-7423", "storage_set": "storage-set-1", "type": "data_node/1"}', '{"machine_id": "fae544140e673b0f2a0d26cb2011516a", "cluster_id": "8ba28deb-4e10-4f07-a32d-ab0d2aed7ffe", "components": [{"name": "utils", "version": "2.0.0-7423"}, {"name": "motr", "services": ["io"], "version": "2.0.0-7423"}, {"name": "hare", "version": "2.0.0-7423"}], "cvg": [{"devices": {"data": ["/dev/sdg", "/dev/sdh"], "metadata": ["/dev/sdf"]}, "name": "cvg-02", "type": "ios"}], "hostname": "cortx-data-g1-1.cortx-data-headless.cortx.svc.cluster.local", "name": "cortx-data-g1-1", "node_group": "ssc-vm-g2-rhev4-3386.colo.seagate.com", "node_id": "cortx-data-g1-1.cortx-data-headless.cortx.svc.cluster.local", "version": "2.0.0-7423", "storage_set": "storage-set-1", "type": "data_node/1"}']}
`

File: py-utils/src/utils/query_deployment/query_deployment.py (hunk 15 above)
Comment (lakshita-jain) [reply py-utils/src/utils/query_deployment/query_deployment.py:89]: I have Pasted the results below

File: py-utils/src/utils/query_deployment/query_deployment.py (hunk 14 above)
Comment (lakshita-jain) [reply py-utils/src/utils/query_deployment/query_deployment.py]: Done

File: py-utils/src/utils/query_deployment/query_deployment.py (hunk 11 above)
Comment (lakshita-jain) [reply py-utils/src/utils/query_deployment/query_deployment.py:64]: Done

File: py-utils/src/utils/query_deployment/query_deployment.py (hunk 9 above)
Comment (lakshita-jain) [reply py-utils/src/utils/query_deployment/query_deployment.py]: Done

File: py-utils/src/utils/query_deployment/query_deployment.py (hunk 8 above)
Comment (lakshita-jain) [reply py-utils/src/utils/query_deployment/query_deployment.py]: Done

File: py-utils/src/utils/query_deployment/query_deployment.py (hunk 10 above)
Comment (lakshita-jain) [reply py-utils/src/utils/query_deployment/query_deployment.py]: Done

File: py-utils/src/utils/query_deployment/query_deployment.py (hunk 13 above)
Comment (lakshita-jain) [reply py-utils/src/utils/query_deployment/query_deployment.py]: Done

File: py-utils/src/utils/query_deployment/query_deployment.py (hunk 14 above)
Comment (lakshita-jain) [reply py-utils/src/utils/query_deployment/query_deployment.py]: Done

File: py-utils/src/utils/query_deployment/query_deployment.py (hunk 15 above)
Comment (lakshita-jain) [reply py-utils/src/utils/query_deployment/query_deployment.py:89]: Done

File: py-utils/src/utils/query_deployment/query_deployment.py (hunk 11 above)
Comment (lakshita-jain) [reply py-utils/src/utils/query_deployment/query_deployment.py:64]: Done

**Review by nitin-seagate — APPROVED** [review]
LGTM

**Review by sachinpunadikar — APPROVED** [review]
+1

**Comment by lakshita-jain** [comment]
Deployement  successful; https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/10281/

**Pull request merged**